implementations/elixir/ockam/ockam/lib/ockam/transport/uds.ex

```diff
@@ -0,0 +1,59 @@
+defmodule Ockam.Transport.UDS do
+  use Supervisor
+
+  alias Ockam.Router
+  alias Ockam.Message
+  alias Ockam.Wire
+  alias Ockam.Transport.UDSAddress
+  alias Ockam.Transport.UDS.{ConnectionSupervisor, Listener}
+
+  def start(opts \\ []) do
+    Router.set_message_handler(
+      UDSAddress.type(),
+      {__MODULE__, :handle_transport_message, []}
+    )
+
+    Supervisor.start_link(__MODULE__, opts)
+  end
+
+  @impl true
+  def init(opts) do
+    children = [
+      ConnectionSupervisor,
+      {Listener, opts}
+    ]
+
+    Supervisor.init(children, strategy: :rest_for_one)
+  end
+
+  def handle_transport_message(message) do
+    with {:ok, destination, message} <-
+           pick_destination_and_set_onward_route(message),
+         {:ok, encoded_message} <- Wire.encode(message),
+         {:ok, socket} <-
+           :gen_tcp.connect({:local, destination}, 0, [:binary, active: true, reuseaddr: true]),
```

Comment:
For connecting to a remote node, if we do that in the message handler, what happens with the connection?
Should we have some "client" worker to manage the connection in that case?